### PR TITLE
1st step: es6 going to be supported

### PR DESCRIPTION
**Pull request by eyang-pjrcorp**

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -317,7 +317,17 @@ async function traverse(directory) {
...
         let extension = path.extname(itemPath).toLowerCase();
-        if (extension === ".js" || extension === ".cjs" || extension === "") {
+        let isCommonJS = false;
+        if (
+          _app.type === "commonjs" &&
+          (extension === ".js" || extension === ".cjs" || extension === "")
+        ) {
+          isCommonJS = true;
+        } else if (_app.type === "module" && extension === ".cjs") {
+          isCommonJS = true;
```
Comment (igibek) [inline src/index.js]: probably, here isCommonJS must be false

Comment (hzeng-pjr) [reply src/index.js]: Friday Evan and I discussed about this commit. I made a flow chart Friday and we are trying to revise/write the code for this flow chart. I will email you the flow chart.

**Review by hzeng-pjr — APPROVED** [review]

File: src/index.js [hunk 2]
```diff
@@ -427,3 +474,15 @@ async function reduceModule(modul, extra = null) {
 
   await Reducer.reduce(modul, additional);
 }
+
+function isValidFile(file) {
```
Comment (hzeng-pjr) [inline src/index.js:478]: suggest to add a comment to describe this function isValidFile(file) . For example, something like the following:
Check whether the files are valid. For now, the files with ".js", ".cjs", ".mjs", and {empty extension but filename is not starting with "."} are valid. Example invalid files according to the last condition are: .eslintc, .editorconfig, .gitattributes, .npmignore, .hidden.

File: src/lib/Configurator.js [hunk 3]
```diff
@@ -67,7 +67,7 @@ settings.silent = argv.silent;
 settings.compressLog = argv.compressLog;
 settings.logOutput = argv.logOutput ? argv.logOutput : settings.logOutput;
 
-settings.ignored = [".git", ".gitignore"];
+settings.ignored = [".git", ".gitignore", "LICENSE", "Makefile", "Make"];
```
Comment (hzeng-pjr) [inline src/lib/Configurator.js:70]: ".git" and ".gitignore"  have already been filtered out in function isValidFile(file) , correct?

**Pull request merged**